app/services/order_items_service.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class OrderItemsService
+  attr_reader :order, :params, :order_item
+
+  DEFAULT_QUANTITY_INCREMENT = 1
+
+  def initialize(order:, params:)
+    @order = order
+    @params = params
+  end
+
+  def call
+    find_order_item ? update_order_item : create_order_item
+  end
+
+  def find_order_item
+    @order_item = order.order_items.find_by(book: params[:book_id])
+  end
+
+  def update_order_item
+    return if user_want_more_book_when_we_have
+
+    order_item.quantity < DEFAULT_QUANTITY_INCREMENT ? order_item.delete : order_item.save
```

Comment:
wild logic, as I got it, if I pass negative value it can delete my order_item - nonsense

Reply:
same